src/models/Tosser.js

```diff
@@ -76,7 +77,16 @@ class Tosser {
         } else if (typeof strategy === 'string') {
           const handlerFun = handler.getRetryHandler(strategy);
           noRetry = handlerFun(ctx);
-        } else {
+        } else if (status) {
+          if (Array.isArray(status)) {
+            noRetry = status.toString().split(",").includes(ctx.res.statusCode.toString());
```

Comment:
I think we need the opposite logic here. We need to retry on given statuses.

```suggestion
            noRetry = !status.includes(this.response.statusCode);
```